canfdtest.c

```diff
@@ -346,7 +350,7 @@ static int can_echo_gen(void)
 
 			/* own frame */
 			if (rx_frame.can_id == can_id_ping) {
-				compare_frame(&tx_frames[recv_tx_pos], &rx_frame, 0);
+				err = compare_frame(&tx_frames[recv_tx_pos], &rx_frame, 0);
```

Comment:
Where is the `err` evaluated?

Reply:
yes, and err is just returned, passed on to main() and there returned on the very last line.